site/src/api/queries/users.ts

```diff
@@ -11,9 +19,12 @@ export const updatePassword = () => {
   };
 };
 
-export const createUser = () => {
+export const createUser = (queryClient: QueryClient) => {
   return {
     mutationFn: API.createUser,
+    onSuccess: async () => {
```

Comment:
On [line 24](https://github.com/coder/coder/pull/9932/files#diff-68be75f92567456822af483809e2cbe7d00dd757dc8b25f35ae98da00cbede90R24), we don't have a closure like we do on [line 8](https://github.com/coder/coder/pull/9932/files#diff-68be75f92567456822af483809e2cbe7d00dd757dc8b25f35ae98da00cbede90R8). Do you know why? Referencing[ this convo](https://codercom.slack.com/archives/CJURPL8DN/p1695992391872739) in Slack the other day.

Reply:
It is because one is a query and another one a mutation so when it is a mutation, you can pass the args on the call like `createUser.mutateAsync(...)`